### PR TITLE
Proposed new function for messy - duplicate rows of data

### PR DESCRIPTION
**Pull request by Philip-Leftwich**
This function, duplicate_rows, takes a dataframe and allows you to duplicate a certain percentage of its rows and reinsert them into the dataframe, either maintaining the original order or inserting them randomly.

Really cool package by the way - that I have already started using to add some mess to my students data!

### CODE REVIEW

File: R/duplicate_rows.R [hunk 1]
```diff
@@ -0,0 +1,63 @@
+#' Duplicate rows and insert them into the dataframe in order or at random
+#'
+#' @param data input dataframe
+#' @param messiness Percentage of rows to duplicate. Must be
+#' between 0 and 1. Default 0.1.
+#' @param shuffle Insert duplicated data underneath original data or insert randomly
+#' @importFrom dplyr slice_sample mutate arrange
```
Comment (nrennie) [inline R/duplicate_rows.R:7]: Please use namespacing not ImportFrom, then delete this line.

File: R/duplicate_rows.R [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+#' @return A dataframe with duplicated rows inserted
+#' @export
+#' @examples
+#' duplicate_rows_random_insert(mtcars, messiness = 0.1)
```
Comment (nrennie) [inline R/duplicate_rows.R:11]: This example doesn't match the function name.

**Comment by nrennie** [comment]
@Philip-Leftwich Thanks so much for this PR - it's definitely another messy data problem that students would benefit from practicing with! Just a couple of small fixes needed:

- Can you put the PR into `dev` rather than `main`?
- Can you update the NEWS file and bump the version number in the DESCRIPTION to 0.0.2.9000?
- Run `devtools::document()` to write the .Rd files for the new function

File: R/duplicate_rows.R [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+  }
+
+  # Calculate the number of rows to duplicate
+  num_rows_to_duplicate <- ceiling(nrow(data) * messiness)
+
+
+  # Add an index column to preserve original order
+  # Mark rows as originals
+  data <- data |>
+    dplyr::mutate(original_index = row_number()) |>
```
Comment (nrennie) [inline R/duplicate_rows.R:24]: add dplyr::

**Review by nrennie — CHANGES_REQUESTED** [review]
@Philip-Leftwich Thanks so much for this PR - it's definitely another messy data problem that students would benefit from practicing with! Just a couple of small fixes needed:

- Can you put the PR into dev rather than main?
- Can you update the NEWS file and bump the version number in the DESCRIPTION to 0.0.2.9000?
- Run `devtools::document()` to write the .Rd files for the new function
- Pass the linting checks with no warnings

**Comment by Philip-Leftwich** [comment]
Hi,
Thanks for doing those changes - I was off sick last week. Was just going to take a look now!

**Pull request merged**